mixer/adapter/redisquota/config/config.proto

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 the Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package adapter.redisquota.config;
+
+import "google/protobuf/duration.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "config";
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+message Params {
+  message Override {
+    option (gogoproto.goproto_getters) = true;
+
+    // The specific dimensions for which this override applies.
+    // String representation of instance dimensions is used to check against configured dimensions.
```

Comment:
Is this by choice or due to constraints in the Mixer's type system?

It would be preferable to have the values here be expressed in a strongly-typed way instead of just strings.

Reply:
I followed the way memquota adapter handles override. 
I am not sure how I can achieve that. Either map<string, istio.mixer.v1.template.Value> or map<string, google.protobuf.Any> ?

Reply:
You're right, there's no way to express this in an adapter config. Ideally, this would be ValueType and we would deliver the adapter the values in a suitable form, something like struct Value {ValueType, interface{}}

@guptasu WDYT?

Reply:
I am not sure I understand the use case of this field. So the use in the adapter config would write something like:

override:
  dimensions:
    "d1": 12345
    "d2: "somestring"
    "d3": true

where d1 is a label of ValueType that is determined by the expression that operator has written in instance config
Is my understanding right ?

Can't this issue be solve if the adapter config has
map <string, google.protobuf.Struct> dimensions = 1

and whatever operator write in adapter config we will deliver to the adapter during SetConfig time.

Then during runtime, we already pass the adapters, the ValueType and the value of the label, which adapter code can use to decide whether to apply the override or not.

Am I missing something

Reply:
I'm trying to get a strongly-typed story end to end. I think google.protobuf.Value will not provide the validation that our type system would, correct? I'd like to follow the same rules as we do for other type inference in config, performing the same error checking, etc.

Reply:
Do we need to include related changes to this PR? Can we do this in the another PR?

Reply:
@mangchiandjjoe yes, we can make the change to the mixer framework separately, outside this PR. Is this adapter getting released, as part of this PR, as a GA or it will be more like a BETA quality and you can make breaking changes later ?

Reply:
@geeknoid PTAL at the bug, https://github.com/istio/istio/issues/2928